src/core/thread/mle.hpp

```diff
@@ -129,22 +129,17 @@ class Mle : public InstanceLocator, private NonCopyable
     /**
      * This method starts the MLE protocol operation.
      *
-     * @param[in]  aAnnounceAttach True if attach on the announced thread network with newer active timestamp,
-     *                             or False if not.
-     *
      * @retval kErrorNone           Successfully started the protocol operation.
      * @retval kErrorInvalidState   IPv6 interface is down or device is in raw-link mode.
      *
      */
-    Error Start(bool aAnnounceAttach);
+    Error Start(void) { return Start(kNormalAttach); }
```

Comment:
Should we update the description of this function? After all, we have removed one parameter and the meaning of the function has also changed.T he following functions are also the same.

Reply:
The PR does already update the documentation for `Start()` (removes the description of the removed `aAnnounceAttach` param). The description "This method starts the MLE protocol operation" is still valid.

@zhanglongxia do you have some other changes in mind?

Reply:
Because we didn't add the comments for the function `Start(StartMode aMode)`. I thought we may add some notes here to remind the reader that this operation won't attach on the announced thread network with newer active timestamp. Thoughts?

Reply:
With this update the public `Start()` simply starts the MLE operation.

How/when the announce messages are handled is all done by `Mle` class itself, so it all internal/private behavior of this class. And users of `Mle` methods don't need to know anything about it. Before this we were using the same `Start()` for both internal to `Mle` and also as a public method (for use by other modules). But with this we made a `Start()` flavor for private use.